### PR TITLE
sys-power/powertop: generate version information for SCM build

### PR DESCRIPTION
**Pull request by eliasp**
Also:
- update header date
- migrate to EAPI=6 and git-r3 eclass

Package-Manager: portage-2.2.28

### CODE REVIEW

File: sys-power/powertop/powertop-9999.ebuild [hunk 1]
```diff
@@ -96,6 +96,8 @@ pkg_setup() {
 
 src_prepare() {
 	if [[ ${PV} == "9999" ]] ; then
+		chmod +x scripts/version || die "Failed to make 'scripts/version' executable"
+		scripts/version || die "Failed to extract version information"
 		eautoreconf
```
Comment (SoapGentoo) [inline sys-power/powertop/powertop-9999.ebuild]: are you sure `src_prepare()` will exit cleanly? where do you call `eapply_user` when `${PV} == "9999"`?

**Comment by gktrk** [comment]
@ZeroChaos-

File: sys-power/powertop/powertop-9999.ebuild (hunk 1 above)
Comment (eliasp) [inline sys-power/powertop/powertop-9999.ebuild]: > are you sure src_prepare() will exit cleanly?
In terms of? I didn't change much when it comes to this and I don't think there's much different here to other ebuilds.

> where do you call eapply_user when ${PV} == "9999"?
Fixed - thanks!

**Comment by eliasp** [comment]
@gktrk Regarding your label `enhancement` - this is actually a bugfix which is required due to upstream changes introduced in https://github.com/fenrus75/powertop/commit/2daaaf54d07340610af759cc7f4c41742abb3277 - otherwise the build will simply fail of current `9999` and future releases.

File: sys-power/powertop/powertop-9999.ebuild [hunk 2]
```diff
@@ -96,6 +96,9 @@ pkg_setup() {
 
 src_prepare() {
```
Comment (SoapGentoo) [inline sys-power/powertop/powertop-9999.ebuild]: write the `src_prepare` like so:

```
src_prepare() {
	default
	if [[ ${PV} == "9999" ]] ; then
		chmod +x scripts/version || die "Failed to make 'scripts/version' executable"
		scripts/version || die "Failed to extract version information"
		eautoreconf
	fi
}
```
This applies `PATCHES` whether `PV` is `9999` or not, and then calls `eautoreconf` in the live ebuild.

Comment (eliasp) [inline sys-power/powertop/powertop-9999.ebuild]: Thanks a lot - this looks much better. Updated the PR accordingly.

**Comment by SoapGentoo** [comment]
@eliasp I just saw, just nix the `DOCS` variable as well, the files included are part of the default set anyways in `EAPI=6` (https://devmanual.gentoo.org/ebuild-writing/functions/src_install/index.html).

**Comment by eliasp** [comment]
@SoapGentoo Thanks - done!

**Comment by jktjkt** [comment]
Looks good to me and I can verify that it fixes the build failure.

**Comment by ZeroChaos-** [comment]
ACK, I hate this, but I don't see anything in the eclass that wraps git describe, so ack.

**Comment by gktrk** [comment]
Merged in https://github.com/gentoo/gentoo/commit/0d66aec25c7d49561549007eec73e17bbe1d997b. Thanks!

**Pull request closed without merging**